apps/verticalhome/js/stores/item.js

```diff
@@ -270,6 +307,83 @@
       });
     },
 
+
+    /**
+     * This function saves all items to DB
+     * @param  {[Array of objects]} objects [Ojects of applications]
+     * @return {[nothins]}
+     */
+    savePinApps: function (objects) {
+      for (var i = 0; i < objects.length; i++) {
+        this.savePinAppItem(objects[i]);
+      }
+    },
+
+    /**
+     * This function save application (item) to DB
+     * @param  {[object (application)]}   object   [Application's object]
+     * @param  {Function} callback [callback that executes after saving object to DB]
+     * @return {[nothing]}
+     */
+    savePinAppItem: function (object, callback) {
+      // intentional use of == meaning null or undefined.
+      if (object.index == null) {
+        console.error('Attempting to save object without `index`');
+        return;
+      }
+
+      newTxn(
+
+              DB_PIN_APP_STORE,
+              'readwrite',
+              function (txn, store) {
+                var request = store.get(object.index);
+                request.onsuccess = function (req) {
+                  var objFromDB = req.result;
+                  if (objFromDB) {
+                    var DBIndex = objFromDB.index;
+                    objFromDB = object;
+                    objFromDB.index = DBIndex;
+                    store.put(objFromDB);
+                  } else {
+                    store.put(object);
+                  }
+                }
+              },
+              callback);
```

Comment:
So this function allows you to blow away existing database entries by just re-using an existing index, it seems.
Wouldn't it be better to use the manifestURL (which is the unique identifier for apps) as the key for this object store? Especially since you're also creating an index for the index property?